### DISCUSSION TITLE
Is it possible to use PHPBench and PHPUnit simultaneously?

### DISCUSSION THREAD

**Discussion by Derison** [phpbench/phpbench#1036] [Q&A]
I am very intersted in monitoring the execution time of my php code. My project has a very high code coverage, so that mostly every php function has a dedicated test in it.
I am thinking of implementing PHPBench now, but I am kinda scared that I need to double my maintenance effort, when I implement it and I need to create both test and an extra bench suit.

I found this issue https://github.com/phpbench/phpbench/issues/210 which seems to do what I want. Just wrapping the benchmarks around the tests. Somehow it doesn't work for me because it complains about the missing PHPUnit test class.

So, is it a good approach to do it like this? Or is it more recommended to always maintain an own bench suit?

**Comment by dantleech** [comment]
There is no best practice here, but I would concentrate on benchmarking things which are performance sensitive, it wouldn't be useful to have f.e. 100% benchmark coverage.

While you can (as of a few years back anyway) use the PHPUnit test cases, the assertions in those tests will also take time and pollute the results. (I think in your case you are not including `runner.bootstrap = vendor/autoload.php`?) but this depends on the scale of the test. If it takes 1 second then, whatever, those assertions won't make any difference, if it's 1ms then they will.

I would have a separate suite.
